Paper task 4: mathematical background (#556)
fpvandoorn/carleson: paper/main.tex

\section{The Metric Space Carleson theorem}\label{sec:main_thm}

\subsection{Mathematical background}\label{subsec:background}

> The basic object of study in Carleson's Theorem is the Fourier series of a periodic function on the real numbers.

Reviewer: ```suggestion
The basic object of study in Carleson's theorem is the Fourier series of a periodic function on the real numbers.
```

> For a function $f : [0, 2\pi) \to \complex$, we first define its Fourier coefficients as

Reviewer: ```suggestion
Given a function $f : [0, 2\pi) \to \complex$ and $n \in \mathbb{Z}$, we first define the $n$th Fourier coefficient of $f$ as
```

\begin{equation}\label{eq:fourier-coefficients}
    c_n \coloneq \widehat{f}_n \coloneq \frac 1{2\pi}\int_0^{2\pi}f(x) e^{- i nx}\, dx.
\end{equation}

> The Lean version \lean{fourierCoeff} uses mathlib's formalization of the Bochner integral and thus is well-defined if $f\in L^1(0,2\pi)$.

Reviewer: ```suggestion
The Lean version \lean{fourierCoeff} uses \mathlib's formalization of the Bochner integral and thus is well-defined if $f\in L^1(0,2\pi)$.
```

We define the $N$-th partial Fourier sum of $f$ at $x\in [0,2\pi]$  as
\begin{equation}\label{eq:trig-series}
    S_N f(x) \coloneq \sum_{n=-N}^N c_n e^{inx} \
\end{equation}
and its Fourier series as $\lim_{N\to\infty} S_N f (x)$.

The metric space Carleson theorem in \cite{CarlesonBlueprint} is a restricted weak type estimate for a generalized Carleson operator. One of the generalizations here is that the domain of the functions operated on can be any \emph{doubling metric measure space} $(X,\rho,\mu, a)$. This is defined to be a complete
and locally compact metric space $(X,\rho)$
equipped with a non-zero locally finite Borel measure $\mu$ that satisfies the following doubling condition for $a\in\nat$.
For all $x\in X$ and all $R>0$ we have
\begin{equation}\label{doublingx}
    \mu(B(x,2R))\le 2^a\mu(B(x,R))\,,
\end{equation}
where $B(x,R):=\{y\in X: \rho(x,y)<R\}$ denotes the open ball of radius $R$ centered at $x$ (see \lean{DoublingMeasure}).
%In a doubling metric measure space the closed balls are compact and $\mu$ is positive on all non-empty open sets.

> The proof of the metric space Carleson theorem relies on several tools from real analysis and part of the formalization is a proof of the Marcinkiewicz real interpolation theorem (see \ref{subsubsec:real_interpolation}). Statement and proof require the following definitions for which we could rely on mathlib's measure theory (see e.g. \cite{vanDoornFormalizedHaarMeasure} for an overview).

Reviewer: ```suggestion
The proof of the metric space Carleson theorem relies on several tools from real analysis and part of the formalization is a proof of the Marcinkiewicz real interpolation theorem (see \ref{subsubsec:real_interpolation}). Statement and proof require the following definitions for which we could rely on \mathlib's measure theory (see e.g. \cite{vanDoornFormalizedHaarMeasure} for an overview).
```

\begin{equation}
  \|f\|_{L^{p,\infty}} \coloneq \sup_{t>0} t \cdot \mu (\{x\in X : t < |f(x)| \}) ^ \frac{1}{p}
\end{equation}
if $0<p<\infty$, and $\|f\|_{L^{\infty,\infty}} \coloneq \|f\|_{L^{\infty}}$.
The function $f$ is said to be in weak $L^p$ (\lean{MemWLp}) if $f$ is $\mu$-a.e. strongly measurable with respect to a fixed topology on $E$ (see e.g. \cite{vanDoornFormalizedHaarMeasure, GouezelChangeOfVariables} for motivation of the different notions of measurability in mathlib) and $\|f\|_{L^{p,\infty}} < \infty$.
An important consequence is that if $f$ is in weak $L^p$, then $|f|$ is finite almost everywhere (see \lean{MemWLp.ae_ne_top}).

Given two measure spaces $(X,\mu)$ and $(Y,\nu)$, and two topological spaces $E$ and $F$ endowed with extended norms, we say that an operator $T : (X \to E) \to (Y \to F)$ is of weak type $(p,p')$ if there exists a constant $C$ such that for any $f\in L^p(X; E)$, we have that $T f$ is $\nu$-a.e. strongly measurable and
\begin{equation}
  \|T f\|_{L^{p',\infty}} \le C \cdot \|f\|_{L^p}.
\end{equation}

> %TODO: should we state the real interpolation theorem here, in the dedicated subsection, or omit the exact formulation and just cite the paper that is to appear about its formalization?

Reviewer: Maybe in the dedicated subsection? I would not include it here.

Author: Ok, I'll add it there then.
